### PR TITLE
[TTAHUB-1215] When gettings goal data for report, we shouldn't check objective status

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
Draft AR's should be able to have draft or complete objectives on them

## How to test
WIth the most recent prod snapshot, using user 146, edit the goal on report 25395. You should not lose any data.
Also fixes jira 1216, instructions  on how to test are in that ticket

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1215
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1216

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good.

File: src/services/goals.js [hunk 1]
```diff
@@ -547,11 +547,6 @@ export async function goalsByIdsAndActivityReport(id, activityReportId) {
                 [Op.ne]: '',
               },
             },
-            {
-              status: {
-                [Op.notIn]: [OBJECTIVE_STATUS.COMPLETE, OBJECTIVE_STATUS.DRAFT],
-              },
-            },
```
[the remark this answers is not in the document]
Comment (thewatermethod) [reply src/services/goals.js:554]: This fetches objectives that are _already on a given report_ not objectives to be added to a report

[the remark this answers is not in the document]
Comment (AdamAdHocTeam) [reply src/services/goals.js:554]: If this is the case I should be good.

**Review by GarrettEHill — APPROVED** [review]
I think this is the change we need now, but need to rethink how we record the data of state change for goals and objectives as the date and their use are important.

**Review by nvms — APPROVED** [review]
went through the scenario in jira - looks good

tested AR form after [3f77fa3](https://github.com/HHS/Head-Start-TTADP/pull/1185/commits/3f77fa3d7faf6d9c86af9405097c638b6f596de4) - looks good

**Pull request merged**